xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -192,6 +192,16 @@ void CGUIWindowPVRGuideBase::UpdateButtons(void)
 
 bool CGUIWindowPVRGuideBase::Update(const std::string &strDirectory, bool updateFilterPath /* = true */)
 {
+  if (m_vecItemsUpdating)
+  {
+    // Prevent concurrent updates. Instead, let the timeline items refresh thread pick it up later.
+    CSingleLock lock(m_critSection);
```

Comment:
why do you need the lock? m_bRefreshTimelineItems is atomic

Reply:
Good point. Will have to check other places as well I guess.